### ISSUE TITLE
Implement slip ring alongside limit switch telescope

### ISSUE THREAD

**Issue by pdnelson** [YCPRadioTelescope/YCP-RT-ControlRoom#165]
Right now, we have limit switches as the only type of Radio Telescope we can use, but we also want to have a slip ring as an option.

~~After the user is asked if they would like to create a new telescope (#164), the user will be asked whether the telescope uses a slip ring or not. This setting will be saved to the database with the new radio telescope, and the user will be able to find this setting again in the "Settings" menu of the "Diagnostics Form" (such a menu will be created in #168).~~

The telescope will be set to "slip ring" by default once development with hard stops concludes.

What slip ring code will involve:
- Bypass azimuth limit switches/stops
  - Also remove the option to use the overrides for those switches
- Calculate shortest travel distance with full 360-degree rotation
- Save slip ring/limit switch setting as an enum in the database (`radio_telescope.telescope_type`)
  - Back end issue: https://github.com/YCPRadioTelescope/RT-Contracts/issues/54
- Communicate with the Mechanical Engineering team with their progress on this

**Comment by pdnelson** [comment]
Continuing my research into this task:
My proposition for calculating the shortest distance: If the azimuth `target_orientation - current_orientation` exceeds 180°, or falls below -180°, it will go the other way, because that path is not the shortest. The furthest the azimuth motor will have to drive the telescope in one movement is 180°/-180°, so anything above that creates a shorter path.

Once a shorter path is detected using the method above, a calculation will exist to drive the telescope in the opposite direction:
The calculation I came up with is: `(x - 360) - y = shorter path`

The way I arrived at this can be viewed [here](https://cdn.discordapp.com/attachments/750831386453213324/779766218617257984/20201121_125158.jpg).
